Kernel/Syscalls/execve.cpp

```diff
@@ -560,9 +560,23 @@ ErrorOr<void> Process::do_exec(NonnullLockRefPtr<OpenFileDescription> main_progr
     m_environment = move(environment);
 
     TRY(m_unveil_data.with([&](auto& unveil_data) -> ErrorOr<void> {
-        unveil_data.state = VeilState::None;
-        unveil_data.paths.clear();
-        unveil_data.paths.set_metadata({ TRY(KString::try_create("/"sv)), UnveilAccess::None, false });
+        TRY(m_exec_unveil_data.with([&](auto& exec_unveil_data) -> ErrorOr<void> {
+            // Note: If we have exec unveil data being waiting to be dispatched
+            // to the current execve'd program, then we apply the unveil data and
+            // ensure it is locked in the new program.
+            if (exec_unveil_data.state == VeilState::Dropped) {
+                unveil_data.paths = TRY(exec_unveil_data.paths.deep_copy());
```

Comment:
If we return early from any of these TRY()s, is the process's unveil data in an inconsistent state? Could someone do some strange things with failed-exec processes as a result? I wonder if we should commit the veil separately from duplicating the data.

Reply:
I changed the order of setting the state and paths now, it should be good now:
```c++
unveil_data.state = VeilState::LockedInherited;
unveil_data.paths = TRY(exec_unveil_data.paths.deep_copy());
```
instead of how it was:
```c++
unveil_data.paths = TRY(exec_unveil_data.paths.deep_copy());
unveil_data.state = VeilState::LockedInherited;
```